docs/contributing.md

```diff
@@ -219,64 +219,67 @@ multi-threaded. It should use all the CPU cores without any configuration.
 
-The documentation for `jj` is automatically published to the website at
+The documentation for `jj` is automatically published online at
 <https://martinvonz.github.io/jj/>.
 
-When editing documentation, we'd appreciate it if you checked that the
-result will look as expected when published to the website.
+When editing documentation, you should check your changes locally — especially
+if you are adding a new page, or doing a major rewrite.
 
-### Setting up the prerequisites
+The only thing you need is [`uv`][uv], a Python project manager written in Rust.
+It will fetch the right Python version and the dependencies. Install it like so:
 
-To build the website, you must have Python and `poetry 1.8+` installed (the
-latest version is recommended). It is easiest to install `poetry` via `pipx`, as
-explained in the [Poetry installation instructions]. A few helpful points from
-the instructions: `pipx` can often be installed from your distribution, e.g.
-`sudo apt install pipx`; this will usually also install Python for you if
-necessary. Any version of `pipx` will do. If you are installing `pipx` manually,
-you may first need to follow the [Python installation instructions].
+[uv]: https://docs.astral.sh/uv/
 
-[Python installation instructions]: https://docs.python.org/3/using/index.html
-[Poetry installation instructions]: https://python-poetry.org/docs/#installation
+=== "macOS/Linux"
 
-Once you have `poetry` installed, you should ask it to install the rest
-of the required tools into a virtual environment as follows:
+    ``` { .shell .copy }
+    curl -LsSf https://astral.sh/uv/install.sh | sh
+    ```
 
-```shell
-poetry install
-```
+=== "Windows"
 
-You may get requests to "unlock a keyring", [an error messages about failing to
-do so](https://github.com/python-poetry/poetry/issues/1917), or Poetry may
-[simply hang indefinitely](https://github.com/python-poetry/poetry/issues/8623).
-The workaround is to either to unlock the keyring or to run the following, and
-then to try `poetry install` again:
+    ``` { .shell .copy }
+    powershell -ExecutionPolicy ByPass -c "irm https://astral.sh/uv/install.ps1 | iex"
+    ```
 
-```shell
-# For sh-compatible shells or recent versions of `fish`
-export PYTHON_KEYRING_BACKEND=keyring.backends.fail.Keyring
-```
+=== "Homebrew"
 
-### Building the HTML docs locally (with live reload)
+    ``` { .shell .copy }
+    brew install uv
+    ```
 
-The HTML docs are built with [MkDocs](https://github.com/mkdocs/mkdocs). After
-following the above steps, you should be able to view the docs by running
+=== "Cargo"
 
-```shell
-# Note: this and all the commands below should be run from the root of
-# the `jj` source tree.
-poetry run -- mkdocs serve
+    ``` { .shell .copy }
+    # This might take a while
+    cargo install --git https://github.com/astral-sh/uv uv
+    ```
+    
+=== "Other options"
+
+    See [Installing uv](https://docs.astral.sh/uv/getting-started/installation/).
```

Comment:
We could add "Download binaries directly from the \[`uv` releases\] page"

Perhaps this could be a list of two items. I'm not sure it's with another tab.